### PR TITLE
Fixed errcheck issues in server/channels/app/import_functions_test.go

### LINKED ISSUE #28949: Fix errcheck issues in server/channels/app/import_functions_test.go
At Mattermost, we are committed to maintaining a high standard of code quality, reliability, and security. To further enhance this, we are launching a community campaign to integrate the [`errcheck`](https://github.com/kisielk/errcheck) linter across the Mattermost codebase.

This Help Wanted ticket is about fixing the linter issues reported by `errcheck` in `channels/app/import_functions_test.go`.

### What is `errcheck`?
`errcheck` is a tool that automatically identifies places in Go code where errors are returned but not checked. In Go, error handling is a critical part of writing robust applications, and ignoring returned errors can lead to bugs, crashes, or unintended behaviour. By integrating `errcheck` into the codebase, we can ensure that error handling is consistently and properly managed across the project.

### Tasks
- Remove the exception for `channels/app/import_functions_test.go` from the exceptions list for `errcheck` in `server/.golangci.yml`.
- Run `make golangci-lint` to get a list of all linter errors.
- Fix the errors by properly handling them.
- Run `make golangci-lint` again to ensure all errors are fixed.

#### Okay, but what does "properly handling" mean?
In general, errors should be returned to the caller. Non critical errors should be loggged, and function execution should continue. There are some exceptions to this rule, such as when the error is expected and should be ignored or when the error is handled in a different way. In such cases, a comment should be added to explain why the error is ignored.

In tests, errors should be checked using `require.NoError(t, err)` for errors of type `error` or `require.Nil(t, err)` for errors of type `**model.AppError`.
If an error is expected, it should be checked using `require.Error(t, err)` for errors of type `error` or `require.NotNil(t, err)` for errors of type `**model.AppError`.

Use `err` as the variable name for errors and `appErr` for `*model.AppError`. Avoid using variable names like `err1`, `err2`, etc. See [the Mattermost style guide](https://developers.mattermost.com/contribute/more-info/server/style-guide/#error-variable-names) for more information.

 https://github.com/mattermost/mattermost/pull/28756 and https://github.com/mattermost/mattermost/pull/28767 can be used as examples of how to fix `errcheck` issues in the `api4` package.

-------------

If you're interested, please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on the Mattermost Community server, where you can discuss questions with community members and the Mattermost core team.
For technical advice or questions, please [join our "Developers Server" community channel](https://community.mattermost.com/core/channels/developers-server).
You can also reach out to @hanzei either on this ticket or on [the Mattermost Community server](https://community.mattermost.com).

For new contributors, please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

If this ticket is already taken, please check out our other [Help Wanted tickets](https://github.com/mattermost/mattermost/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+errcheck+label%3A%22Up+For+Grabs%22) of this campaign.

Thank you for your contributions to Mattermost!

----------------

JIRA: MM-61258

### PR DESCRIPTION
**Pull request by edlerd**
#### Summary

Fixed errcheck issues in server/channels/app/import_functions_test.go

#### Ticket Link

Fixes #28949

### CODE REVIEW

**Review by hanzei — CHANGES_REQUESTED** [review]
Nice PR :+1: 

There are some linter errors to clean up:
```
Error: channels/app/import_functions_test.go:74:2: calling require.NotNil on error, please use require.Error instead
Error: channels/app/import_functions_test.go:83:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:92:2: calling require.NotNil on error, please use require.Error instead
Error: channels/app/import_functions_test.go:101:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:158:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:214:2: calling require.NotNil on error, please use require.Error instead
Error: channels/app/import_functions_test.go:262:2: calling require.NotNil on error, please use require.Error instead
Error: channels/app/import_functions_test.go:271:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:280:2: calling require.NotNil on error, please use require.Error instead
Error: channels/app/import_functions_test.go:289:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:346:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:402:2: calling require.NotNil on error, please use require.Error instead
Error: channels/app/import_functions_test.go:626:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:643:2: calling require.NotNil on error, please use require.Error instead
Error: channels/app/import_functions_test.go:652:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:660:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:668:2: calling require.NotNil on error, please use require.Error instead
Error: channels/app/import_functions_test.go:677:2: calling require.NotNil on error, please use require.Error instead
Error: channels/app/import_functions_test.go:691:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:698:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:715:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:722:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:741:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:743:2: calling require.Nil on error, please use require.NoError instead
Error: channels/app/import_functions_test.go:5581:5: calling require.NoError on *github.com/mattermost/mattermost/server/public/model.AppError, please use require.Nil instead
Error: channels/app/import_functions_test.go:5589:5: calling require.NoError on *github.com/mattermost/mattermost/server/public/model.AppError, please use require.Nil instead
Error: channels/app/import_functions_test.go:5734:5: calling require.NoError on *github.com/mattermost/mattermost/server/public/model.AppError, please use require.Nil instead
```



Please let me know if you need any help

**Comment by edlerd** [comment]
Updated the linter issued and ensured `make vet` passes locally without issues. Maybe add it as a description to the other "Fix errcheck issues ..." descriptions. Because I ensured `errcheck` is ok, but didn't expect make vet to find more issues.
Please give another review.

File: server/channels/app/import_functions_test.go [hunk 1]
```diff
@@ -900,22 +910,26 @@ func TestImportImportUser(t *testing.T) {
 
 	// Test team and channel memberships
 	teamName := model.NewRandomTeamName()
-	th.App.importTeam(th.Context, &imports.TeamImportData{
+	err = th.App.importTeam(th.Context, &imports.TeamImportData{
```
Comment (hanzei) [inline server/channels/app/import_functions_test.go:913]: `th.App.importTeam` returns a `*model.AppError`. You can't assign that to an `error ` variable.

Comment (edlerd) [reply server/channels/app/import_functions_test.go]: ah right, got it. thx for the hint.

**Comment by edlerd** [comment]
test cases and make vet passes locally :heavy_check_mark:

**Comment by hanzei** [comment]
Let's see if CI is also happy

File: server/channels/app/import_functions_test.go [hunk 2]
```diff
@@ -5202,66 +5256,75 @@ func TestZippedImportPostAndRepliesWithAttachments(t *testing.T) {
 
 	// Create a Team.
 	teamName := model.NewRandomTeamName()
-	th.App.importTeam(th.Context, &imports.TeamImportData{
+	err := th.App.importTeam(th.Context, &imports.TeamImportData{
```
Comment (hanzei) [inline server/channels/app/import_functions_test.go:5259]: Can this be `appErr`?

File: server/channels/app/import_functions_test.go [hunk 3]
```diff
@@ -5202,66 +5256,75 @@ func TestZippedImportPostAndRepliesWithAttachments(t *testing.T) {
...
 	testsDir, _ := fileutils.FindDir("tests")
 	testImage := filepath.Join(testsDir, "test.png")
 	testZipFileName := filepath.Join(testsDir, "import_test.zip")
 	testZip, _ := os.Open(testZipFileName)
 
-	fi, err := testZip.Stat()
-	require.NoError(t, err, "failed to get file info")
-	testZipReader, err := zip.NewReader(testZip, fi.Size())
-	require.NoError(t, err, "failed to read test zip")
+	fi, fiErr := testZip.Stat()
```
Comment (hanzei) [inline server/channels/app/import_functions_test.go:5324]: Can we keep this variable name as `err`?

File: server/channels/app/import_functions_test.go [hunk 4]
```diff
@@ -5551,8 +5640,16 @@ func BenchmarkCompareFilesContent(b *testing.B) {
...
+
+		defer func() {
+			err = zipFile.Close()
+			require.NoError(b, err)
+		}()
+
+		defer func() {
+			err = os.Remove(zipFilePath)
+			require.NoError(b, err)
+		}()
```
Comment (hanzei) [inline server/channels/app/import_functions_test.go:5652]: Nit: We can combine these two functions

File: server/channels/app/import_functions_test.go [hunk 5]
```diff
@@ -5631,11 +5728,19 @@ func BenchmarkCompareFilesContent(b *testing.B) {
 
 			_, appErr := th.App.WriteFile(zipFile, "compareFiles.zip")
 			require.Nil(b, appErr)
-			defer th.App.RemoveFile("compareFiles.zip")
+
```
Comment (hanzei) [inline server/channels/app/import_functions_test.go:5731]: Please remove the new line

**Review by hanzei — CHANGES_REQUESTED** [review]
Great work @edlerd :tada: 

A few comments below

**Comment by edlerd** [comment]
Not sure why the enterprise tests are failing. I don't have access to the results and the details link is 404 for me. Are those failures maybe unrelated to the changes introduced here?

**Comment by hanzei** [comment]
Your branch is out f sync with the closed source code. I've merged `master` into your branch. That should fix the check.

**Comment by edlerd** [comment]
@hanzei the merge worked, but ci needs to be triggered again and I guess this might be ready to merge after it passes.

**Comment by hanzei** [comment]
@edlerd Sorry, @edlerd, there is yet another merge conflict. Can you please take a look? Thanks!

**Comment by edlerd** [comment]
I rebased on main and resolved the conflicts. @hanzei please run ci and merge as it passes.

**Review by hanzei — APPROVED** [review]
Great work! There are some unresolved comments, but none of them is blocking. e.g. https://github.com/mattermost/mattermost/pull/29229/files#r1841724285

**Review by BenCookie95 — APPROVED** [review]
LGTM, thanks @edlerd

**Comment by edlerd** [comment]
Great, this was a big one with quite a few learnings for me. Thx for the reviews and suggestions 😀

**Pull request merged**